ContactsManagerBackend/ContactsManager/ContactsManagerDAL/ContactsProvider.cs

```diff
@@ -38,5 +34,153 @@ public List<Contact> GetContacts()
 
             return Contacts;
         }
+
+        public Contact GetContact(Guid Id)
+        {
+            if (Id == null)
+            {
+                throw new ArgumentException();
+            }
+
+            var contact = new Contact();
+
+            using (SqlConnection conn = DBConnection.GetSqlConnection())
+            using (SqlCommand cmd = conn.CreateCommand())
+            {
+                cmd.CommandText = "getContactById";
+                cmd.CommandType = CommandType.StoredProcedure;
+
+                cmd.Parameters.Add("Id", SqlDbType.UniqueIdentifier);
```

Comment:
Add returns the parameter. So, you can do this ".Add(...).Value = Id".
Create an extension method that does this, and takes all three parameters